### PR TITLE
Add tests examples and improve documentation for WebSocket

### PR DESCRIPTION
**Pull request by riyafa**

### CODE REVIEW

File: stdlib/http/src/main/ballerina/http/Module.md [hunk 1]
```diff
@@ -51,21 +57,24 @@ There are also two types of services for WebSocket: `WebSocketService` and `WebS
     }
 }
 ```
-Here `upgradeService` is a `WebSocketService`.
+Here `upgradeService` is a server callback service.
```
Comment (praneesha) [inline stdlib/http/src/main/ballerina/http/Module.md]:
```suggestion
The `upgradeService` is a server callback service.
```

File: stdlib/http/src/main/ballerina/http/Module.md [hunk 2]
```diff
@@ -51,21 +57,24 @@ There are also two types of services for WebSocket: `WebSocketService` and `WebS
...
 
-**onOpen resource**: As soon as the WebSocket handshake is completed and the connection is established, the `onOpen` resource is dispatched. This resource is only available in the `WebSocketService`.
+**onOpen resource**: As soon as the WebSocket handshake is completed and the connection is established, the `onOpen` 
+resource is dispatched. This resource is only available in the server service.
```
Comment (praneesha) [inline stdlib/http/src/main/ballerina/http/Module.md]:
```suggestion
resource is dispatched. This resource is only available in the service of the server.
```

File: stdlib/http/src/main/ballerina/http/Module.md [hunk 3]
```diff
@@ -51,21 +57,24 @@ There are also two types of services for WebSocket: `WebSocketService` and `WebS
...
 
 **onText resource**: Received Text messages are dispatched to this resource.
 
 **onBinary resource**: Received Binary messages are dispatched to this resource.
 
-**onPing and onPong resources**: Received ping and pong messages are dispatched to these resources respectively as a `blob`
+**onPing and onPong resources**: Received ping and pong messages are dispatched to these resources respectively.
```
Comment (praneesha) [inline stdlib/http/src/main/ballerina/http/Module.md]:
```suggestion
**onPing and onPong resources**: The received ping and pong messages are dispatched to these resources respectively.
```

File: stdlib/http/src/main/ballerina/http/Module.md [hunk 4]
```diff
@@ -51,21 +57,24 @@ There are also two types of services for WebSocket: `WebSocketService` and `WebS
...
 
-**onIdleTimeout**: This resource is dispatched when idle timeout is reached. idleTimeout has to be configured by the user in WebSocket service configuration.
+**onIdleTimeout**: This resource is dispatched when idle timeout is reached. idleTimeout has to be configured by the
```
Comment (praneesha) [inline stdlib/http/src/main/ballerina/http/Module.md]:
```suggestion
**onIdleTimeout**: This resource is dispatched when the idle timeout is reached. idleTimeout has to be configured by the 
```

File: stdlib/http/src/main/ballerina/http/Module.md [hunk 5]
```diff
@@ -51,21 +57,24 @@ There are also two types of services for WebSocket: `WebSocketService` and `WebS
...
 **onText resource**: Received Text messages are dispatched to this resource.
 
 **onBinary resource**: Received Binary messages are dispatched to this resource.
 
-**onPing and onPong resources**: Received ping and pong messages are dispatched to these resources respectively as a `blob`
+**onPing and onPong resources**: Received ping and pong messages are dispatched to these resources respectively.
 
-**onIdleTimeout**: This resource is dispatched when idle timeout is reached. idleTimeout has to be configured by the user in WebSocket service configuration.
+**onIdleTimeout**: This resource is dispatched when idle timeout is reached. idleTimeout has to be configured by the 
+user in WebSocket service or client configuration.
```
Comment (praneesha) [inline stdlib/http/src/main/ballerina/http/Module.md]:
```suggestion
user in the WebSocket service or client configuration.
```

File: stdlib/http/src/main/ballerina/http/Module.md [hunk 6]
```diff
@@ -51,21 +57,24 @@ There are also two types of services for WebSocket: `WebSocketService` and `WebS
...
 
 **onClose**: This resource is dispatched when a close message is received.
```
Comment (praneesha) [inline stdlib/http/src/main/ballerina/http/Module.md]: It is not clear what a close message is.

File: stdlib/http/src/main/ballerina/http/annotation.bal [hunk 7]
```diff
@@ -77,8 +77,9 @@ public type Versioning record {
 # + endpoints - An array of endpoints the service would be attached to
 # + path - Path of the WebSocket service
 # + subProtocols - Negotiable sub protocol by the service
-# + idleTimeoutInSeconds - Idle timeout for the client connection. This can be triggered by putting
-#                          an `onIdleTimeout` resource in the WebSocket service.
+# + idleTimeoutInSeconds - Idle timeout for the client connection. Upon timeout, `onIdleTimeout` resource (if defined)
+#                          in the server service will be triggered. Note that this overrides the `timeoutMillis` config
+#                          in the `http:Listener`.
 # + maxFrameSize - The maximum payload size of a WebSocket frame in bytes.
 #                  If this is not set or is negative or zero the default frame size of 65536 will be used.
```
Comment (praneesha) [inline stdlib/http/src/main/ballerina/http/annotation.bal]:
```suggestion
#                  If this is not set or is negative or zero, the default frame size of 65536 will be used.
```
Do we need to mention the measurement unit of 65536?

File: stdlib/http/src/main/ballerina/http/annotation.bal [hunk 8]
```diff
@@ -126,10 +127,10 @@ public type HttpResourceConfig record {
     !...
 };
 
-# Configures the HTTP to WebSocket upgrade.
+# Configuration for HTTP to WebSocket upgrade.
```
Comment (praneesha) [inline stdlib/http/src/main/ballerina/http/annotation.bal]:
```suggestion
# Configuration to upgrade WebSocket to HTTP.
```

File: stdlib/http/src/main/ballerina/http/annotation.bal [hunk 9]
```diff
@@ -126,10 +127,10 @@ public type HttpResourceConfig record {
...
 #
 # + upgradePath - Path which is used to upgrade from HTTP to WebSocket
-# + upgradeService - WebSocket service which should be used after a successful upgrade
+# + upgradeService - Callback service for a successful upgrade
```
Comment (praneesha) [inline stdlib/http/src/main/ballerina/http/annotation.bal]:
```suggestion
# + upgradeService - Callback service for a successful upgrade.
```

File: stdlib/http/src/main/ballerina/http/websocket/websocket_caller.bal [hunk 10]
```diff
@@ -17,7 +17,7 @@
 # Represents a WebSocket caller.
 #
 # + id - The connection id
-# + negotiatedSubProtocol - The subprotocols negoriated with the server
+# + negotiatedSubProtocol - The subprotocols negotiated with the server
```
Comment (praneesha) [inline stdlib/http/src/main/ballerina/http/websocket/websocket_caller.bal]:
```suggestion
# + negotiatedSubProtocol - The subprotocols that are negotiated with the server.
```

File: stdlib/http/src/main/ballerina/http/websocket/websocket_caller.bal [hunk 11]
```diff
@@ -34,7 +34,7 @@ public type WebSocketCaller client object {
     # Push text to the connection.
     #
     # + data - Data to be sent, if byte[] it is converted to a UTF-8 string for sending
-    # + finalFrame - True if this is a final frame of a (long) message
+    # + finalFrame - `true` if this is a final frame of a (long) message
```
Comment (praneesha) [inline stdlib/http/src/main/ballerina/http/websocket/websocket_caller.bal]:
```suggestion
    # + finalFrame - Set to `true` if this is a final frame of a (long) message
```

File: stdlib/http/src/main/ballerina/http/websocket/websocket_caller.bal [hunk 12]
```diff
@@ -69,7 +69,7 @@ public type WebSocketCaller client object {
     #
     # + statusCode - Status code for closing the connection
     # + reason - Reason for closing the connection
-    # + timeoutInSecs - Time to waits for the close frame from the remote endpoint before closing the connection.
+    # + timeoutInSecs - Time to wait for the close frame from the remote endpoint before closing the connection.
```
Comment (praneesha) [inline stdlib/http/src/main/ballerina/http/websocket/websocket_caller.bal]:
```suggestion
    # + timeoutInSecs - Time to wait for the close frame being received from the remote endpoint before closing the connection.
```

File: stdlib/http/src/main/ballerina/http/websocket/websocket_client.bal [hunk 13]
```diff
@@ -17,7 +17,7 @@
 # Represents a WebSocket client endpoint.
 #
 # + id - The connection id
-# + negotiatedSubProtocol - The subprotocols negoriated with the server
+# + negotiatedSubProtocol - The subprotocols negotiated with the server
```
Comment (praneesha) [inline stdlib/http/src/main/ballerina/http/websocket/websocket_client.bal]:
```suggestion
# + negotiatedSubProtocol - The subprotocols that are negotiated with the server
```

File: stdlib/http/src/main/ballerina/http/websocket/websocket_client.bal [hunk 14]
```diff
@@ -35,7 +35,7 @@ public type WebSocketClient client object {
     private string url = "";
     private WebSocketClientEndpointConfig config = {};
 
-    # Gets called when the endpoint is being initialize during module init time.
+    # Gets called to initialize the client.
```
Comment (praneesha) [inline stdlib/http/src/main/ballerina/http/websocket/websocket_client.bal]:
```suggestion
    # Initializes the client when called.

File: stdlib/http/src/main/ballerina/http/websocket/websocket_client.bal [hunk 15]
```diff
@@ -50,7 +50,7 @@ public type WebSocketClient client object {
     # Push text to the connection.
     #
     # + data - Data to be sent, if byte[] it is converted to a UTF-8 string for sending
-    # + finalFrame - True if this is a final frame of a (long) message
+    # + finalFrame - `true` if this is a final frame of a (long) message
```
Comment (praneesha) [inline stdlib/http/src/main/ballerina/http/websocket/websocket_client.bal]:
```suggestion
    # + finalFrame - Set to `true` if this is a final frame of a (long) message
```

File: stdlib/http/src/main/ballerina/http/websocket/websocket_client.bal [hunk 16]
```diff
@@ -59,7 +59,7 @@ public type WebSocketClient client object {
     # Push binary data to the connection.
     #
     # + data - Binary data to be sent
-    # + finalFrame - True if this is a final frame of a (long) message
+    # + finalFrame - `true` if this is a final frame of a (long) message
```
Comment (praneesha) [inline stdlib/http/src/main/ballerina/http/websocket/websocket_client.bal]:
```suggestion
    # + finalFrame - Set to `true` if this is a final frame of a (long) message
```

File: stdlib/http/src/main/ballerina/http/websocket/websocket_client.bal [hunk 17]
```diff
@@ -85,7 +85,7 @@ public type WebSocketClient client object {
     #
     # + statusCode - Status code for closing the connection
     # + reason - Reason for closing the connection
-    # + timeoutInSecs - Time to waits for the close frame from the remote endpoint before closing the connection.
+    # + timeoutInSecs - Time to wait for the close frame from the remote endpoint before closing the connection.
```
Comment (praneesha) [inline stdlib/http/src/main/ballerina/http/websocket/websocket_client.bal]:
```suggestion
    # + timeoutInSecs - Time to wait for the close frame to be received from the remote endpoint before closing the connection.
```

File: stdlib/http/src/main/ballerina/http/websocket/websocket_client.bal [hunk 18]
```diff
@@ -85,7 +85,7 @@ public type WebSocketClient client object {
...
     #                   If the timeout exceeds then the connection is terminated even though a close frame
```
Comment (praneesha) [inline stdlib/http/src/main/ballerina/http/websocket/websocket_client.bal]:
```suggestion
    #                   If the timeout exceeds, then the connection is terminated even though a close frame
```

File: stdlib/http/src/main/ballerina/http/websocket/websocket_client.bal [hunk 19]
```diff
@@ -85,7 +85,7 @@ public type WebSocketClient client object {
...
     #                   is not received from the remote endpoint. If the value < 0 (eg: -1) the connection waits
```
Comment (praneesha) [inline stdlib/http/src/main/ballerina/http/websocket/websocket_client.bal]:
```suggestion
    #                   is not received from the remote endpoint. If the value < 0 (e.g., -1), then the connection waits
```

File: stdlib/http/src/main/ballerina/http/websocket/websocket_client.bal [hunk 20]
```diff
@@ -85,7 +85,7 @@ public type WebSocketClient client object {
...
     #                   until a close frame is received. If WebSocket frame is received from the remote endpoint
```
Comment (praneesha) [inline stdlib/http/src/main/ballerina/http/websocket/websocket_client.bal]:
```suggestion
    #                   until a close frame is received. If WebSocket frame is received from the remote endpoint,
```

File: stdlib/http/src/main/ballerina/http/websocket/websocket_client.bal [hunk 21]
```diff
@@ -104,13 +104,15 @@ public type WebSocketClient client object {
     }
 };
 
-# Configuration struct for WebSocket client endpoint.
+# Configuration for WebSocket client endpoint.
```
Comment (praneesha) [inline stdlib/http/src/main/ballerina/http/websocket/websocket_client.bal]:
```suggestion
# Configuration for the WebSocket client endpoint
```

File: stdlib/http/src/main/ballerina/http/websocket/websocket_connector.bal [hunk 22]
```diff
@@ -16,7 +16,7 @@
 
 import ballerina/internal;
 
-# Represents a WebSocket connector in ballerina. This include all connection oriented operations.
+# Represents a WebSocket connection in ballerina. This includes all connection oriented operations.
```
Comment (praneesha) [inline stdlib/http/src/main/ballerina/http/websocket/websocket_connector.bal]:
```suggestion
# Represents a WebSocket connection in Ballerina. This includes all connection-oriented operations.
```

File: stdlib/http/src/main/ballerina/http/websocket/websocket_connector.bal [hunk 23]
```diff
@@ -52,7 +52,7 @@ type WebSocketConnector object {
     # Push binary data to the connection.
     #
     # + data - Binary data to be sent
-    # + finalFrame - True if this is a final frame of a (long) message
+    # + finalFrame - `true` if this is a final frame of a (long) message
```
Comment (praneesha) [inline stdlib/http/src/main/ballerina/http/websocket/websocket_connector.bal]:
```suggestion
    # + finalFrame - Set to`true` if this is a final frame of a (long) message
```

File: stdlib/http/src/main/ballerina/http/websocket/websocket_connector.bal [hunk 24]
```diff
@@ -72,7 +72,7 @@ type WebSocketConnector object {
     #
     # + statusCode - Status code for closing the connection
     # + reason - Reason for closing the connection
-    # + timeoutInSecs - Time to waits for the close frame from the remote endpoint before closing the connection.
+    # + timeoutInSecs - Time to wait for the close frame from the remote endpoint before closing the connection.
```
Comment (praneesha) [inline stdlib/http/src/main/ballerina/http/websocket/websocket_connector.bal]:
```suggestion
    # + timeoutInSecs - Time to wait for the close frame to be received from the remote endpoint before closing the connection.
```

File: stdlib/http/src/main/ballerina/http/websocket/websocket_connector.bal [hunk 25]
```diff
@@ -72,7 +72,7 @@ type WebSocketConnector object {
...
     #                   If the timeout exceeds then the connection is terminated even though a close frame
     #                   is not received from the remote endpoint. If the value < 0 (eg: -1) the connection waits
```
Comment (praneesha) [inline stdlib/http/src/main/ballerina/http/websocket/websocket_connector.bal]:
```suggestion
    #                   is not received from the remote endpoint. If the value < 0 (e.g., -1), then the connection waits
```

File: stdlib/http/src/main/ballerina/http/websocket/websocket_caller.bal [hunk 26]
```diff
@@ -69,7 +69,7 @@ public type WebSocketCaller client object {
...
     #                   If the timeout exceeds then the connection is terminated even though a close frame
     #                   is not received from the remote endpoint. If the value < 0 (eg: -1) the connection waits
```
Comment (praneesha) [inline stdlib/http/src/main/ballerina/http/websocket/websocket_caller.bal]:
```suggestion
    #                   is not received from the remote endpoint. If the value < 0 (e.g., -1) the connection waits
```

File: stdlib/http/src/main/ballerina/http/Module.md (hunk 6 above)
Comment (riyafa) [reply stdlib/http/src/main/ballerina/http/Module.md]: Fixed. Please check.

File: stdlib/http/src/main/ballerina/http/annotation.bal (hunk 7 above)
Comment (riyafa) [reply stdlib/http/src/main/ballerina/http/annotation.bal]: Removed it.

File: stdlib/http/src/main/ballerina/http/annotation.bal (hunk 8 above)
Comment (riyafa) [reply stdlib/http/src/main/ballerina/http/annotation.bal]: Changed this. Please verify.

**Pull request merged**